external/js/kyber/src/@types/elliptic/index.d.ts

```diff
@@ -0,0 +1,62 @@
+// types declaration for version 6.4.1 for Kyber usage
+//
+// The definition contains only a subset of the library
```

Comment:
Not a criticism, just to understand: do you expect these local @types files to go away as you manage to get the upstreams to take your fixes to their .d files?

Reply:
So that's the main reason why I don't like TypeScript, most of the library you need don't have a types definition. You can find decent definitions on [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) but to take the example of bn.js, their version was totally wrong regarding how the library works and _elliptic_ is incomplete.
I decided to write the definitions for the project because fixing the ones on DefinitelyTyped would require some time. Up to you if you want me to do that to use them eventually.

Reply:
Yes, we always try to contribute to projects which we depend on and we notice a problem that we have to work around. It makes everything better, for the other users, and our own code has fewer weird things in it that we have to maintain.